consensus/tendermint/tendermint.go

```diff
@@ -294,17 +301,17 @@ func (t *Tendermint[V, H, A]) startRound(r round) Action[V, H, A] {
 }
 
 type timeout struct {
-	s step
-	h height
-	r round
+	S step   `cbor:"s"`
+	H height `cbor:"h"`
+	R round  `cbor:"r"`
```

Comment:
```suggestion
	_ struct{} `cbor:",toarray"`
	S step
	H height
	R round
```
The default encoding mode is using a map with the field name as key. I think it would be much more efficient to use the array mode instead. In that case, it's not required to annotate the fields, as the field name doesn't matter anymore.

Reply:
Sample
```
package main

import (
	"encoding/hex"
	"fmt"

	"github.com/fxamacker/cbor/v2"
)

type A1 struct {
	X int
	Y int
	Z string
}

type B1 struct {
	A1
	U int
	V string
}

type A2 struct {
	_ struct{} `cbor:",toarray"`
	X int
	Y int
	Z string
}

type B2 struct {
	_ struct{} `cbor:",toarray"`
	A2
	U int
	V string
}

func main() {
	b1 := B1{A1: A1{X: 1, Y: 2, Z: "3"}, U: 4, V: "5"}
	b2 := B2{A2: A2{X: 1, Y: 2, Z: "3"}, U: 4, V: "5"}
	res1, _ := cbor.Marshal(b1)
	res2, _ := cbor.Marshal(b2)
	fmt.Println(hex.EncodeToString(res1))
	fmt.Println(hex.EncodeToString(res2))
}
```

```
a5615801615902615a613361550461566135
8501026133046135
```

Reply:
Yup, nice one. Updated